File: 0739-daily-temperatures/0739-daily-temperatures.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        vector<int> answer(temperatures.size()); 
        stack<pair<int,int>> stack; 
        stack.push({temperatures[0],0}); 
        for(int i = 1; i < temperatures.size(); i++){
            int curr = temperatures[i]; 
            while(!stack.empty() && curr > stack.top().first){
                answer[stack.top().second] = i - stack.top().second; 
                stack.pop(); 
            }
            stack.push({curr,i}); 
        }
        return answer; 
    }
};

File: 0739-daily-temperatures/0739-daily-temperatures.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        Deque<int[]> stack = new ArrayDeque<>(); 
        int[] answer = new int[temperatures.length]; 
        
        for(int i = 0; i < temperatures.length; i++){
            int curr = temperatures[i]; 
            if(!stack.isEmpty()){
                while(!stack.isEmpty() && curr > stack.peek()[1]){
                    answer[stack.peek()[0]] = i - stack.peek()[0]; 
                    stack.pop(); 
                }
            }
            stack.push(new int[]{i, curr});
        }
        return answer; 
    }
}